omnigibson/utils/usd_utils.py

```diff
@@ -203,78 +235,89 @@ def initialize_view(cls):
 
         # If there are no valid objects, clear the view and terminate early
-        if i == 0:
-            cls._CONTACT_VIEW = None
+        if len(column_filters) == 0:
+            self._CONTACT_VIEW = dict()
             return
 
+        # Get the row filters too
+        row_filters = self.get_row_filters()
+
         # Generate rigid body view, making sure to update the simulation first (without physics) so that the physx
         # backend is synchronized with any newly added objects
         # We also suppress the omni tensor plugin from giving warnings we expect
         og.sim.pi.update_simulation(elapsedStep=0, currentTime=og.sim.current_time)
         with suppress_omni_log(channels=["omni.physx.tensors.plugin"]):
-            cls._CONTACT_VIEW = og.sim.physics_sim_view.create_rigid_contact_view(
-                pattern="/World/*/*",
-                filter_patterns=list(cls._PATH_TO_COL_IDX.keys()),
-            )
+            for scene_idx, _ in enumerate(og.sim.scenes):
+                # TODO: How to make this work with the floor plane?
+                # If there are no collidable objects in the scene, skip.
+                if len(column_filters[scene_idx]) == 0:
+                    continue
+
+                self._CONTACT_VIEW[scene_idx] = og.sim.physics_sim_view.create_rigid_contact_view(
+                    pattern=row_filters[scene_idx],
+                    filter_patterns=column_filters[scene_idx],
+                    max_contact_data_count=self.get_max_contact_data_count(),
+                )
 
-        # Create deterministic mapping from path to row index
-        cls._PATH_TO_ROW_IDX = {path: i for i, path in enumerate(cls._CONTACT_VIEW.sensor_paths)}
+                self._PATH_TO_ROW_IDX[scene_idx] = {
+                    path: i for i, path in enumerate(self._CONTACT_VIEW[scene_idx].sensor_paths)
+                }
 
-        # Store the reverse mappings as well. This can just be a numpy array since the mapping uses integer indices
-        cls._ROW_IDX_TO_PATH = np.array(list(cls._PATH_TO_ROW_IDX.keys()))
-        cls._COL_IDX_TO_PATH = np.array(list(cls._PATH_TO_COL_IDX.keys()))
+                self._ROW_IDX_TO_PATH[scene_idx] = np.array(list(self._PATH_TO_ROW_IDX[scene_idx].keys()))
+                self._COL_IDX_TO_PATH[scene_idx] = np.array(list(self._PATH_TO_COL_IDX[scene_idx].keys()))
 
         # Sanity check generated view -- this should generate square matrices of shape (N, N, 3)
-        n_bodies = len(cls._PATH_TO_COL_IDX)
-        assert cls._CONTACT_VIEW.filter_count == n_bodies, (
-            f"Got unexpected contact view shape. Expected: (N, {n_bodies}); "
-            f"got: (N, {cls._CONTACT_VIEW.filter_count})"
-        )
+        # n_bodies = len(cls._PATH_TO_COL_IDX)
+        # assert cls._CONTACT_VIEW.filter_count == n_bodies, \
+        #     f"Got unexpected contact view shape. Expected: (N, {n_bodies}); " \
+        #     f"got: (N, {cls._CONTACT_VIEW.filter_count})"
```

Comment:
why remove?

Reply:
@cgokmen

Reply:
You can unremove but make sure you test it. Do update the comment and check the other dimension too.